src/main/java/at/hannibal2/skyhanni/features/misc/MagicalPowerDisplay.kt

```diff
@@ -0,0 +1,106 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.events.RenderItemTipEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.test.command.ErrorManager
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemCategory
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalNameOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getItemCategoryOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getItemRarityOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzRarity
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import io.github.moulberry.notenoughupdates.util.stripControlCodes
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object MagicalPowerDisplay {
+    private val config get() = SkyHanniMod.feature.inventory.magicalPower
+    private var contactAmount: Int
+        get() = ProfileStorageData.profileSpecific?.abiphoneContactAmount ?: -1
+        private set(value) { ProfileStorageData.profileSpecific?.abiphoneContactAmount = value }
+
+    /*
+    * REGEX-TEST: Accessory Bag
+    * REGEX-TEST: Accessory Bag (1/75)
+    * REGEX-TEST: Accessory Bag (909/394294)
+    * REGEX-TEST: Auctions Browser
+    * REGEX-TEST: Auctions: "ligma"
+    * REGEX-TEST: Auctions: ""sugoma""
+    * */
+    private val acceptedInvPattern by RepoPattern.pattern(
+        "inv.acceptable",
+        """^(Accessory Bag(?: \(\d+/\d+\))?|Auctions Browser|Manage Auctions|Auctions: ".*"?)$"""
+    )
+
+    @SubscribeEvent
+    fun onRenderItemTip(event: RenderItemTipEvent) {
+        if (!isEnabled()) return
+        if (!acceptedInvPattern.matches(InventoryUtils.openInventoryName().stripControlCodes())) return
+
+        val item = event.stack
+        val rarity = item.isAccessory() ?: return
+        val itemID = item.getInternalNameOrNull() ?: return
+
+        var endMP = rarity.toMP() ?: run {
+            ErrorManager.skyHanniError(
+                "Unknown rarity '$rarity' for item '${item.displayName}§7'"
+            )
+        }
+        if (itemID == "HEGEMONY_ARTIFACT".asInternalName())
+            endMP *= 2
+        if (itemID == "RIFT_PRISM".asInternalName())
+            endMP = 11
+        if (item.isAbicase())
+            endMP += contactAmount / 2
```

Comment:
Save the InternalNames above. Saves a bit of resources (.asInternalName creates a new object every time)
'''Kotlin
    private val hegemonyArtefact = "HEGEMONY_ARTIFACT".asInternalName()
    private val riftPrism = "RIFT_PRISM".asInternalName()
'''
Also a bit cleaner with a when. I also don't know what to do with the last if, but this way it should be fine.
```suggestion
       when(itemID){
            hegemonyArtefact -> endMP *= 2
            riftPrism -> endMP = 11
            else -> if (item.isAbicase()) {
                endMP += contactAmount / 2
            }
        }
```